### PR TITLE
Support literals cast to aggregates as async reset reg init values

### PR DESCRIPTION
**Pull request by jackkoenig**
Accomplished by changing the code gen for casting literals to
aggregates. Rather than connecting the literal to a wire that is then
bit selected from, just bit select from the literal which saves the
creation of an intermediate wire and matches FIRRTL's semantics for
legal async reset initial values.

Turned out to be a very clean implementation, most work (as usual) in the testing.

**Related issue**: https://github.com/freechipsproject/firrtl/pull/1201

**Type of change**: bug fix

**Impact**: API addition? (only in that it fixes a bug)

**Development Phase**: implementation

**Release Notes**
Fix bug where literals cast to aggregates could not be used as async reset register initial values.

### CODE REVIEW

File: src/test/scala/chiselTests/AsyncResetSpec.scala [hunk 1]
```diff
@@ -169,4 +169,34 @@ class AsyncResetSpec extends ChiselFlatSpec {
...
+      }
+      val (count, done) = Counter(true.B, 4)
+      when (count === 0.U) {
+        chisel3.assert(reg.asUInt === 0xdeadbeefL.U)
+      } .otherwise {
+        chisel3.assert(reg.asUInt === 0xbad0cad0L.U)
+      }
+      when (done) { stop() }
+    })
+    assertTesterPasses(new BasicTester {
```
Comment (ducky64) [inline src/test/scala/chiselTests/AsyncResetSpec.scala]: Suggestion: make this its own test case.

**Comment by johnsbrew** [comment]
@jackkoenig regarding your comment on https://github.com/freechipsproject/firrtl/pull/1201 I guess you do not expect to fix the issues raised in https://github.com/freechipsproject/chisel3/pull/1208 test-cases now but rather wait for an in-depth work about proper constant propagation ?
As far as I understand the current PR does not address the idea of 
```scala
      val x = WireInit(123.U + 456.U)
      withReset(reset.asAsyncReset)(RegInit(x))
```
seen as a valid reset value ?

**Comment by jackkoenig** [comment]
@johnsbrew, you are correct that this is not intended to address test cases like `123.U + 456.U` as a reset value. I'll continue discussion of the more general issue on https://github.com/freechipsproject/firrtl/pull/1201.

File: src/test/scala/chiselTests/AsyncResetSpec.scala (hunk 1 above)
Comment (jackkoenig) [reply src/test/scala/chiselTests/AsyncResetSpec.scala]: 👍

**Comment by jackkoenig** [comment]
Failing CircleCI will be fixed by https://github.com/freechipsproject/chisel3/pull/1227

**Comment by azidar** [comment]
@Mergifyio backport 3.2.x

**Pull request merged**